openpype/pipeline/context_tools.py

```diff
@@ -306,6 +306,58 @@ def ls():
     return host
 
 
+def get_current_host_name():
+    """Current host name.
+
+    Function is based on currently registered host integration or environment
+    variant 'AVALON_APP'.
+
+    Returns:
+        Union[str, None]: Name of host integration in current process or None.
+    """
+
+    host = registered_host()
+    if host is not None and hasattr(host, "name"):
+        return host.name
+    return os.environ.get("AVALON_APP")
+
+
+def get_global_context():
+    return {
+        "project_name": os.environ.get("AVALON_PROJECT"),
+        "asset_name": os.environ.get("AVALON_ASSET"),
+        "task_name": os.environ.get("AVALON_TASK"),
+    }
+
+
+def get_current_context():
+    host = registered_host()
+    if host is not None and hasattr(host, "get_current_context"):
+        return host.get_current_context()
+    return get_global_context()
+
+
+def get_current_project_name():
+    host = registered_host()
+    if host is not None and hasattr(host, "get_current_project_name"):
```

Comment:
I would personally prefer:
```
if isinstance(host, HostBase):
    host.get_current_context()
```
but thats just my preference, couldn't find anything that this would be far superior to `hasattr` (just a bit less cluttered and not depending on a rigid method name, but I guess if wrapper name would need to be changed, we would need to change method names on the class).

And I would like docstring to highlight it coms from Host or from global context for code assist.